tests/prefetch_related/tests.py

```diff
@@ -1510,3 +1510,45 @@ def test_retrieves_results_from_prefetched_objects_cache(self):
         with self.assertNumQueries(4):
             # AuthorWithAge -> Author -> FavoriteAuthors, Book
             self.assertQuerysetEqual(authors, ['<AuthorWithAge: Rousseau>', '<AuthorWithAge: Voltaire>'])
+
+    def test_related_manager_remove_prefetched_objects(self):
```

Comment:
These tests belong with `Model.refresh_from_db()` (tests/basic/tests.py). Try to make them as simple as possible. It looks like the `fields is None` branch isn't tested.